src/backend/distributed/commands/index.c

```diff
@@ -228,13 +227,14 @@ PreprocessIndexStmt(Node *node, const char *createIndexCommand)
 
 
 /*
- * ErrorIfCannotGenerateDefaultIndexName errors out if we cannot generate a
- * default index name for given CREATE INDEX command.
+ * ErrorIfCreateIndexHasTooManyColumns errors out if given CREATE INDEX command
+ * would use more than INDEX_MAX_KEYS columns.
  */
 static void
-ErrorIfCannotGenerateDefaultIndexName(IndexStmt *createIndexStatement)
+ErrorIfCreateIndexHasTooManyColumns(IndexStmt *createIndexStatement)
 {
-	if (CanGenerateDefaultIndexName(createIndexStatement))
+	int numberOfIndexParameters = GetNumberOfIndexParameters(createIndexStatement);
+	if (numberOfIndexParameters <= INDEX_MAX_KEYS)
```

Comment:
Also removed `numberOfIndexParameters == 0` check here as it doesn't make much sense, or should I bring it back ?